fictadvisor-web/src/app/(main)/schedule/schedule-page/calendar-section/components/groups-dropdown/GroupsDropDown.tsx

```diff
@@ -15,23 +16,30 @@ export const GroupsDropDown: FC<DropDownSectionProps> = ({ groups }) => {
   }));
   const setGroupId = useSetGroupId();
 
+  const memoGroups = useMemo(() => {
+    return groups
+      ? groups?.map(group => ({
+          id: group.id,
+          label: group.code,
+        }))
+      : [];
+  }, [groups.length]);
```

Comment:
just groups instead of [groups.length], is inside of the useMemo hook you check for existing a groups

Reply:
groups is an object, so every time it gets recreated a new object will be returned from useMemo, and so it will create the same issue with re-rendenders of DropDown component

Reply:
inside that hook you check for existing a groups (groups ? ... : ...) so wont there be a problem like ```cannot read properties of undefined. reading property "length"``` because groups object is undefined?

Reply:
<img width="482" alt="image" src="https://github.com/user-attachments/assets/36d77c1d-d3aa-4bd8-84ab-fcd9ae157822">
It's not marked it type of this variable. So i cannot be undefined

Reply:
if so, then check for groups.length inside that hook, because now it doesnt make any sense